### PR TITLE
[base] plain pickling support

### LINKED ISSUE #16: Implement pickle for serialization

### PR DESCRIPTION
**Pull request by marscher**
If users restore an object stored with an old version,
they receive a warning that things could fail.

Fixes #16

### CODE REVIEW

File: sktime/markovprocess/__init__.py [hunk 1]
```diff
@@ -4,6 +4,9 @@
 ureg.define('step = []')  # dimensionless unit for unspecified lag time unit.
 Q_ = ureg.Quantity
 
+# TODO: we need to do this for unpickling, but it will overwrite other apps default registry!
```
Comment (clonker) [inline sktime/markovprocess/__init__.py]: huh well that sucks, serious design flaw in pint imo

File: sktime/markovprocess/_dtraj_stats.py [hunk 2]
```diff
@@ -288,6 +285,8 @@ def effective_count_matrix(self):
 
 class TransitionCountEstimator(Estimator):
 
+    # TODO: ctor?
```
Comment (clonker) [inline sktime/markovprocess/_dtraj_stats.py]: i think `dt_traj` could be a ctor argument and should potentially be part of the model... :eyes:

File: sktime/markovprocess/markov_state_model.py [hunk 3]
```diff
@@ -19,19 +19,18 @@
...
 from pyemma.util.linalg import mdot
 
 from sktime.base import Model
-from sktime.markovprocess.sample import _ensure_dtraj_list
-from sktime.util import ensure_ndarray
 from sktime.markovprocess import Q_
 from sktime.markovprocess._dtraj_stats import TransitionCountModel
 from sktime.markovprocess.pcca import PCCA
+from sktime.markovprocess.sample import _ensure_dtraj_list
+from sktime.util import ensure_ndarray
```
Comment (clonker) [inline sktime/markovprocess/markov_state_model.py]: maybe we can lump all these `ensure_...` methods together into one file

File: tests/base/test_pickling.py [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+
+import numpy as np
+
+from sktime.markovprocess import MaximumLikelihoodMSM
+
+
+class TestBase(unittest.TestCase):
+
+    def test_pickle_msm(self):
+        msm = MaximumLikelihoodMSM(lagtime=1, dt_traj='4 ps')
```
Comment (clonker) [inline tests/base/test_pickling.py]: nice! perhaps we can automatize this test a little by creating a method that takes an instance of a model (the test's parameter) so wed just have to have a model factory (`estimator type->model`) which is i think also useful for other tests

File: sktime/markovprocess/_dtraj_stats.py (hunk 2 above)
Comment (marscher) [reply sktime/markovprocess/_dtraj_stats.py]: it is part of the model, but passed on later. The separation/passing around of model/estimator parameters is a bit of a mess across the different estimators.

File: sktime/markovprocess/markov_state_model.py (hunk 3 above)
Comment (marscher) [reply sktime/markovprocess/markov_state_model.py]: will do

File: sktime/markovprocess/__init__.py (hunk 1 above)
Comment (clonker) [reply sktime/markovprocess/__init__.py]: but thats no better then a temporary fix right? at some point there is going to be a model storing things which aren't dimensionless

**Comment by marscher** [comment]
The problem only arises, when one uses a unit, which isn't defined in 
the registry used to unpickle it.

File: sktime/markovprocess/__init__.py (hunk 1 above)
Comment (marscher) [reply sktime/markovprocess/__init__.py]: The problem only arises, when one uses a unit, which isn't defined in the registry used to unpickle it. I think it is very unlikely, that people will be using non-standard time units.

File: tests/base/test_pickling.py (hunk 4 above)
Comment (marscher) [reply tests/base/test_pickling.py]: Most of the time models are created in a very specific way in order to test certain things. Can you please elaborate a bit where you think this factory might be of good use?

File: sktime/markovprocess/__init__.py (hunk 1 above)
Comment (clonker) [reply sktime/markovprocess/__init__.py]: Fair enough :)

**Pull request merged**